storagemarketadapter/client_adapter.go

```diff
@@ -0,0 +1,326 @@
+package storagemarketadapter
+
+// this file implements storagemarket.StorageClientNode
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/prometheus/common/log"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/address"
+	"github.com/filecoin-project/lotus/chain/events"
+	"github.com/filecoin-project/lotus/chain/market"
+	"github.com/filecoin-project/lotus/chain/stmgr"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/lib/cborutil"
+	"github.com/filecoin-project/lotus/node/impl/full"
+	"github.com/filecoin-project/lotus/storagemarket"
+)
+
+type ClientNodeAdapter struct {
+	full.StateAPI
+	full.ChainAPI
+	full.MpoolAPI
+
+	sm *stmgr.StateManager
+	cs *store.ChainStore
+	fm *market.FundMgr
+	ev *events.Events
+}
+
+type clientApi struct {
+	full.ChainAPI
+	full.StateAPI
+}
+
+func NewClientNodeAdapter(state full.StateAPI, chain full.ChainAPI, mpool full.MpoolAPI, sm *stmgr.StateManager, cs *store.ChainStore, fm *market.FundMgr) storagemarket.StorageClientNode {
```

Comment:
this all looks good to me. we should however write up a diff of how these node adapters started vs how they turned out.

they aren't in spec cause they are really unique to this shared module as something meant to be used by multiple implementation.

@anorth et all are meeting to discuss storage miner implementation and it may be worth them looking at our lessons from doing this.